src/resolvelib/resolvers/criterion.py

```diff
@@ -27,17 +19,10 @@
     State,
     build_iter_view,
 )
+from .abstract import AbstractResolver, Result
```

Comment:
The implementations(Resolution/Resolver) should belong to a separate module, with the name like `resolvers/impl.py`